### ISSUE TITLE
Signing service misleading info about established connection

### ISSUE THREAD

**Issue by Jakub89** [golemfactory/concent#774]
We have blockade in middleman, for only one connection at the time, but from `signing_service` point of view it looks like everything works fine when we try to run couple `signing_services` simultaneously.
```
12:06:07 INFO     | Connecting to 127.0.0.1:9055.
12:06:07 INFO     | Connection established.
```
In reality the connection is not established and we need to refactor in some way:
```
    def _connect(self, tcp_socket: socket.socket) -> None:
        """ Creates socket and connects to given HOST and PORT. """
        if self.current_reconnect_delay is not None:
            logger.info(f'Waiting {self.current_reconnect_delay} before connecting.')
            sleep(self.current_reconnect_delay)

        logger.info(f'Connecting to {self.host}:{self.port}.')
        tcp_socket.connect((self.host, self.port))
        logger.info(f'Connection established.')

        # Reset delay on successful connection and set flag that connection is established.
        self.current_reconnect_delay = None
        self._handle_connection(tcp_socket)
```
So that we get 'Connection blocked` and `signing_service` gets closed.

**Comment by PaweuB** [comment]
Pivotal:
https://www.pivotaltracker.com/story/show/160197649

**Comment by rwrzesien** [comment]
As discussed, the problem is that `writer.close()` at the `MiddleMan` side is not handled on `SigningService` side, thus it doesn't know that the connection was immediately closed. This can be resolved in one of the following ways:
- ignore it. The connection was estabilished, but closed immediately no longer exists, but it will not be revealed until `SigningService` will try to send something over socket.
- send something from `SigningService` to `MiddleMan` at the beginning of connection. From what I have found out this is the only way to verify if socket connection is still alive. This will require handling that 'something' on the `MiddleMan` side too, because with first connection `MiddleMan` will receive it (it can ie. ignore it).
- send something from `MiddleMan` to `SigningService`, to let it know that connection was closed. Like the previous point, but reversed, it should be handled by `SigningService`.

This might be also helpful:
https://github.com/python/asyncio/issues/466
